tgui/packages/tgui/interfaces/Sensors.tsx

```diff
@@ -0,0 +1,598 @@
+import { BooleanLike } from '../../common/react';
+import { useBackend } from '../backend';
+import { Box, Button, Section, Table, ProgressBar, Slider } from '../components';
+import { NtosWindow } from '../layouts';
+import { round, clamp } from 'common/math';
+import { Color } from 'common/color';
+
+export type SensorsData = {
+  viewing: BooleanLike;
+  muted: BooleanLike;
+  grid_x: number;
+  grid_y: number;
+  x: number;
+  y: number;
+  direction: number;
+  is_ship: BooleanLike;
+  on: BooleanLike;
+  range: number;
+  health: number;
+  max_health: number;
+  deep_scan_name: string;
+  deep_scan_range: number;
+  deep_scan_toggled: BooleanLike;
+  heat: number;
+  critical_heat: number;
+  status: string;
+  desired_range: number;
+  range_choices: number[];
+  contacts: ContactData[];
+
+  id_on: BooleanLike;
+  id_status: string;
+  id_class: string;
+  id_name: string;
+  can_change_class: BooleanLike;
+  can_change_name: BooleanLike;
+  contact_details: string;
+
+  distress_beacons: DistressBeaconData[];
+  distress_range: number;
+
+  datalink_requests: { name: string; ref: string }[];
+  datalinked: { name: string; ref: string }[];
+};
+
+type ContactData = {
+  name: string;
+  ref: string;
+  bearing: number;
+  can_datalink: BooleanLike;
+  landed: BooleanLike;
+  distance: number;
+  x: number;
+  y: number;
+  color: string; // assigned in ts
+};
+
+type DistressBeaconData = {
+  caller: string;
+  sender: string;
+  bearing: number;
+};
+
+const SensorSection = function (act, data: SensorsData) {
+  const range_choice_max = data.range_choices[data.range_choices.length - 1];
+
+  return (
+    <Section
+      title="Sensor Array Control"
+      buttons={
+        <Button
+          content={
+            'Sector Map View ' + (data.viewing ? 'Engaged' : 'Disengaged')
+          }
+          onClick={() => act('viewing')}
+        />
+      }>
+      <Table>
+        <Table.Row>
+          <Table.Cell>State:</Table.Cell>
+          <Table.Cell>
+            {data.on ? 'Active' : 'Inactive'}
+            {', '}
+            <Button content="on/off" onClick={() => act('toggle')} />
```

Comment:
This should just be one button, like so:
`<Button content={data.on ? "Active" : "Inactive"} color={data.on ? "good" : "bad"} icon={data.on ? "on" : "off"} onClick={() => act('toggle')}>

Reply:
done